src/Agent.Worker/ExecutionContext.cs

```diff
@@ -421,6 +424,19 @@ public void InitializeJob(Pipelines.AgentJobRequestMessage message, Cancellation
                 Container = null;
             }
 
+            // Sidecar Containers
+            if (message.JobSidecarContainers?.Count > 0)
```

Comment:
i don't think `jobSidecarContainers` will ever be null, double check your webapi code.

Reply:
It was not null when the build did not include `services` so I removed the check